### DISCUSSION TITLE
Idiomatic way to save a specific datapoint externally and check it later

### DISCUSSION THREAD

**Discussion by yoav-lavi** [statelyai/xstate#2534] [Q&A]
Hey,

I'm working on a statechart that involves saving a user preference (after clicking a specific button) that would need to be checked at the beginning of the statechart on later executions to prevent anything from running (sort of an opt-out mechanism). Using an action  to save the preference to `localStorage` seems more or less idiomatic as an effect, but checking that same preference in a guard seems a bit out of place. Is there an idiomatic way to do this? I'd rather not have half of the operation within the state (saving) and half without (checking and sending an event / setting an initial context).

Thanks!

**Comment by ChrisShank** [comment]
Could you share some code, I just want to make sure I understand this part "I'd rather not have half of the operation within the state (saving) and half without (checking and sending an event / setting an initial context)."?
